### PR TITLE
Fix metrics CORS problems

### PR DESCRIPTION
**Pull request by n0tl3ss**
This pull requests addresses the https://github.com/micronaut-projects/micronaut-micrometer/issues/397. There is an issue when the CORS OPTIONS requests are fired the requests attributes were not populated and the metrics were dumping the raw value of the path instead of templated one. 

This pull requests adds attributes into OPTIONS request if there is anyRoute that can be matched with different HTTP verb. There is also a test that covers that scenario.

### CODE REVIEW

**Comment by yawkat** [comment]
doing this in the cors filter seems like a hack, especially since non-cors options requests can exist too and cause the same issue. I'd much rather have a fix in RoutingInboundHandler

**Review by yawkat — APPROVED** [review]

**Review by graemerocher — APPROVED** [review]

**Comment by yawkat** [comment]
@timyates points out that this test is still failing: https://github.com/Robin479/micronaut-bugs/blob/main/http-metrics/src/test/java/de/ooch/micronaut/bugs/RunawayClientMetricsTest.java

Looking at https://github.com/micronaut-projects/micronaut-micrometer/issues/397 , it seems like the client test should be a separate issue and can be addressed in a separate PR (though I'm not 100% sure we can fix it in a sane way)

**Review by timyates — APPROVED** [review]

**Comment by graemerocher** [comment]
@yawkat I discussed with @n0tl3ss that this can only really be addressed in the client through new APIs that allow passing a template. He will write a proposal for a future feature to solve this use case.

**Comment by yawkat** [comment]
why is static analysis failing? there's no sonar bot info

**Comment by timyates** [comment]
It's an external branch so doesn't get the keys...

One sec, I'll push a "local" branch

**Comment by timyates** [comment]
@yawkat Sonar says https://github.com/micronaut-projects/micronaut-core/pull/7354/files#diff-9af0f6dd8d3c36389a0ff250f900fd204afbe901b57f4e2d23028ecdad048c56R457 should be parameterized

https://github.com/micronaut-projects/micronaut-core/pull/7361#issuecomment-1125964922

**Comment by n0tl3ss** [comment]
> @yawkat Sonar says https://github.com/micronaut-projects/micronaut-core/pull/7354/files#diff-9af0f6dd8d3c36389a0ff250f900fd204afbe901b57f4e2d23028ecdad048c56R457 should be parameterized
> 
> [#7361 (comment)](https://github.com/micronaut-projects/micronaut-core/pull/7361#issuecomment-1125964922)

is changing `HttpRequest` to `HttpRequest<?>` sufficient enough?

**Comment by yawkat** [comment]
yes

**Comment by n0tl3ss** [comment]
> yes

Just added it :) Thanks guys

**Comment by timyates** [comment]
Cool, re-running in the local branch 👍

**Comment by timyates** [comment]
image

💚

**Comment by yawkat** [comment]
Thanks for the help @timyates !

File: http-server-netty/src/test/groovy/io/micronaut/http/server/netty/OptionsRequestAttributesSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import io.micronaut.runtime.server.EmbeddedServer
+import jakarta.inject.Singleton
+import org.reactivestreams.Publisher
+import org.spockframework.util.Assert
+import spock.lang.Specification
+
+
+class OptionsRequestAttributesSpec extends Specification {
+
+    def 'test OPTIONS rquests attributes'() {
```
Comment (donbeave) [inline http-server-netty/src/test/groovy/io/micronaut/http/server/netty/OptionsRequestAttributesSpec.groovy:26]: Typo in a word `rquests`

**Comment by olvesh** [comment]
Sorry, it doesn't look like this is completely fixed:
Running version 3.7.0 of Micronaut I see this when getting a 401
image

 

Trying a GET against a controller not configured for GET (and a auth filter before)
Could this be alleviated by some change at our end (not really experienced in Micronaut specifics I must say)

**Comment by n0tl3ss** [comment]
Please try to use `micronaut-micrometer` version v4.8.2. It should be solved. Thanks :)

**Comment by olvesh** [comment]
Thanks, that did the trick!

**Pull request merged**